File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/game/CardCountList.java
package com.shuneault.netrunnerdeckbuilder.game;

import java.util.ArrayList;

public class CardCountList extends ArrayList<CardCount> {
    public CardCount getCardCount(Card card) {
        for (int i = 0; i < this.size(); i++) {
            if (this.get(i).getCard().getCode().equals(card.getCode())){
                return this.get(i);
            }
        }

        return null;
    }

    public void setCount(Card card, int count) {
        CardCount cardCount = this.getCardCount(card);
        if (cardCount != null){
            cardCount.setCount(count);
        }
        else
        {
            this.add(new CardCount(card, count));
        }
    }
}

File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/game/CardCountList.kt
package com.shuneault.netrunnerdeckbuilder.game
import java.util.ArrayList

class CardCountList : ArrayList<CardCount?>() {
    fun getCardCount(card: Card): CardCount? {
        for (i in this.indices) {
            if (this[i]!!.card.code == card.code) {
                return this[i]
            }
        }
        return null
    }

    fun setCount(card: Card, count: Int) {
        val cardCount = getCardCount(card)
        if (cardCount != null) {
            cardCount.count = count
        } else {
            this.add(CardCount(card, count))
        }
    }
}